piptools/utils.py

```diff
@@ -678,13 +678,24 @@ def parse_config_file(
             hint=f"Could not parse '{config_file !s}': {value_err !s}",
         )
 
-    # In a TOML file, we expect the config to be under `[tool.pip-tools]`
+    # In a TOML file, we expect the config to be under `[tool.pip-tools]`,
+    # `[tool.pip-tools.compile]` or `[tool.pip-tools.sync]`
     piptools_config: dict[str, Any] = config.get("tool", {}).get("pip-tools", {})
+
+    # TODO: Replace with `str.removeprefix()` once dropped 3.8
+    assert click_context.command.name is not None
+    config_section_name = click_context.command.name.replace("pip-", "")
```

Comment:
I think this
```suggestion
    config_section_name = click_context.command.name[len("pip-"):]
```
would be a more accurate representation of what `str.removeprefix()` does.